corehq/motech/fhir/tasks.py

```diff
@@ -394,5 +398,252 @@ def create_parent_indices(
+                        break
+                reason_code = appointment_resource.get('reasonCode')
+                if reason_code and reason_code[0] is not None:
+                    reason = reason_code[0].get('text')
+            host_case_id = patient.get_case_property('case_id')
+            appointment_case_data = {
+                'case_name': appointment_fhir_timestamp + appointment_description,
+                'case_type': 'appointment',
+                'indices': {
+                    'patient': {
+                        'case_id': host_case_id,
+                        'case_type': 'patient',
+                        'relationship': 'extension',
+                    }
+                },
+                'properties': {
+                    'appointment_description': appointment_description,
+                    'appointment_fhir_timestamp': appointment_fhir_timestamp,
+                    'appointment_date': appointment_date,
+                    'appointment_time': appointment_time,
+                    'patient_fhir_id': patient_fhir_id,
+                    'fhir_id': appointment_fhir_id,
+                    'reason': reason,
+                    'practitioner': practitioner
+                }
+            }
+            appointment_create_helper.create_case(appointment_case_data)
+
+        # Update existing appointments in commcare if properties have changed in epic
+        for appointment in epic_appointments_to_update:
+            epic_properties_map = {}
+            appointment_resource = appointment.get('resource')
+            if appointment_resource is not None:
+                appointment_description = appointment_resource.get('description') or 'NO DESCRIPTION LISTED'
+                appointment_fhir_timestamp = appointment_resource.get('start')
+                appointment_date, appointment_time = convert_utc_timestamp_to_date_and_time(
+                    appointment_fhir_timestamp)
+                appointment_fhir_id = appointment_resource.get('id')
+                reason = None
+                practitioner = None
+                for p in appointment_resource.get('participant'):
+                    actor = p.get('actor')
+                    if actor and 'Practitioner' in actor.get('reference'):
+                        practitioner = actor.get('display')
+                        break
+                reason_code = appointment_resource.get('reasonCode')
+                if reason_code and reason_code[0] is not None:
+                    reason = reason_code[0].get('text')
+                epic_properties_map.update({
+                    'appointment_description': appointment_description,
+                    'appointment_fhir_timestamp': appointment_fhir_timestamp,
+                    'practitioner': practitioner,
+                    'reason': reason
+                })
+            appointment_case = None
+            for case in appointment_cases:
```

Comment:
since you're already iterating over these above, you could create a dict mapping the FHIR ID to the case to make this lookup easier.